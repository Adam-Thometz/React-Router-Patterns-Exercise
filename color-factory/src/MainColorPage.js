import React from "react";
import {Link} from 'react-router-dom'

const MainColorPage = () => {
  const colors = Object.keys(localStorage).reverse()
  return (
    <div className="MainColorPage">
      <div className="MainColorPage-header">
        <h1>Welcome to the Color Factory!</h1>
        <h3><Link to='/colors/new'>Add a color</Link></h3>
      </div>
      <div className="MainColorPage-body">
        <p>Please select a color</p>
        {colors.length ? 
          colors.map(c => (
            <p><Link to={`/colors/${c}`} >{c}</Link></p>
          )) :
          <small>You haven't picked any colors yet. Please click <Link to="/colors/new">here</Link> to add colors!</small>}
      </div>
    </div>
  )
}

export default MainColorPage