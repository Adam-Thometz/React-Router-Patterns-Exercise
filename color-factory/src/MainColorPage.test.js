import { render, screen } from '@testing-library/react';
import MainColorPage from './MainColorPage';
import {MemoryRouter} from 'react-router-dom'

test('renders component', () => {
  render(<MemoryRouter>
    <MainColorPage />
  </MemoryRouter>);
});

test('matches snapshot', () => {
  const {asFragment} = render(<MemoryRouter>
    <MainColorPage />
  </MemoryRouter>)
  expect(asFragment()).toMatchSnapshot()
})