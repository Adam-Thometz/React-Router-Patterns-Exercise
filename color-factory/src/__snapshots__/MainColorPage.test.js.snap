// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MainColorPage"
  >
    <div
      class="MainColorPage-header"
    >
      <h1>
        Welcome to the Color Factory!
      </h1>
      <h3>
        <a
          href="/colors/new"
        >
          Add a color
        </a>
      </h3>
    </div>
    <div
      class="MainColorPage-body"
    >
      <p>
        Please select a color
      </p>
      <small>
        You haven't picked any colors yet. Please click 
        <a
          href="/colors/new"
        >
          here
        </a>
         to add colors!
      </small>
    </div>
  </div>
</DocumentFragment>
`;
